### PR TITLE
Upgrade compiler to v0.60.x and tcgc in Sep.

### PR DESCRIPTION
**Pull request by v-jiaodi**
Upgrade compiler version to `v0.60.0`
Upgrade tcgc version to `v0.46.1`

Error log:
```
Invoking: tsc -p .
src/modular/buildCodeModel.ts(1385,34): error TS2339: Property 'values' does not exist on type 'SdkUnionType<SdkType>'.
src/modular/buildCodeModel.ts(1385,46): error TS7006: Parameter 'x' implicitly has an 'any' type.
src/modular/buildCodeModel.ts(1412,31): error TS7006: Parameter 'x' implicitly has an 'any' type.
src/modular/emitSamples.ts(11,3): error TS2724: '"@azure-tools/typespec-client-generator-core"' has no exported member named 'SdkHttpParameterExample'. Did you mean 'SdkHttpParameterExampleValue'?
src/modular/emitSamples.ts(15,3): error TS2305: Module '"@azure-tools/typespec-client-generator-core"' has no exported member 'SdkTypeExample'.
src/modular/emitSamples.ts(448,42): error TS7006: Parameter 'element' implicitly has an 'any' type.
src/utils/credentialUtils.ts(76,47): error TS2339: Property 'values' does not exist on type 'SdkUnionType<SdkCredentialType>'.

```

### CODE REVIEW

File: packages/typespec-test/test/confidentialLedger/generated/openapi/2022-05-13/openapi.json [hunk 1]
```diff
@@ -206,7 +206,8 @@
           }
         },
         "x-ms-pageable": {
-          "nextLinkName": "nextLink"
+          "nextLinkName": "nextLink",
```
Comment (MaryGao) [inline packages/typespec-test/test/confidentialLedger/generated/openapi/2022-05-13/openapi.json:209]: could you check relevant pr change?

File: packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/responses.ts [hunk 2]
```diff
@@ -17,13 +17,13 @@ export interface UsersCreate200Response extends HttpResponse {
   body: UserCreatedResponseOutput;
 }
 
-/** The request conflicts with the current state of the server. */
+/** The user already exists */
```
Comment (MaryGao) [inline packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/responses.ts:20]: Could you help check the relevant pr?

File: packages/typespec-test/test/confidentialLedger/generated/openapi/2022-05-13/openapi.json (hunk 1 above)
Comment (v-jiaodi) [reply packages/typespec-test/test/confidentialLedger/generated/openapi/2022-05-13/openapi.json:209]: pr link: https://github.com/Azure/typespec-azure/pull/1449/files

File: packages/typespec-test/test/anomalyDetector/generated/openapi/v1.1/openapi.json [hunk 3]
```diff
@@ -148,7 +148,8 @@
           }
         },
         "x-ms-pageable": {
-          "nextLinkName": "nextLink"
+          "nextLinkName": "nextLink",
+          "itemName": "models"
```
Comment (MaryGao) [inline packages/typespec-test/test/anomalyDetector/generated/openapi/v1.1/openapi.json:152]: Should be fixed here: https://github.com/Azure/typespec-azure/pull/1449/files

File: packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/responses.ts (hunk 2 above)
Comment (MaryGao) [reply packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/responses.ts:20]: Didn't find relevant pr yet, but this seems a bugfix and both apply to js and openapi emitters.

**Review by MaryGao — APPROVED** [review]

File: packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/responses.ts (hunk 2 above)
Comment (MaryGao) [reply packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/responses.ts:20]: > Didn't find relevant pr yet, but this seems a bugfix and both apply to js and openapi emitters.

@v-jiaodi thanks for di's sharing and should be here: https://github.com/microsoft/typespec/pull/4322.

**Comment by MaryGao** [comment]
Hold on until the issue is fixed: https://github.com/Azure/typespec-azure/issues/1532.

File: packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/models/models.ts [hunk 4]
```diff
@@ -687,3 +687,11 @@ export interface UnivariateUnivariateChangePointDetectionResult {
 
 /** Type of Versions */
 export type Versions = "v1.1";
+
+/** Error response */
+export interface MultivariateResponseError {
```
Comment (MaryGao) [inline packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/models/models.ts:692]: why this model is generated?

Comment (MaryGao) [reply packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/models/models.ts:692]: issue tracked here: https://github.com/Azure/typespec-azure/issues/1587

**Pull request merged**